### PR TITLE
Fix CMake warnings + drop old style export of library

### PR DESCRIPTION
**Pull request by N-Dekker**
Fixed CMake warnings reported by Jean-Christophe Fillion-Robin (jcfr) at #21
Dropped old style export of Elastix library files.

### CODE REVIEW

File: UseElastix.cmake.in [hunk 1]
```diff
@@ -14,13 +14,8 @@ include_directories( ${ELASTIX_INCLUDE_DIRS} )
 link_directories( ${ELASTIX_LIBRARY_DIRS} )
 
 # If Elastix_FOUND is set, this file is included via find_package() which provides
-# ELASTIX_CONFIG_TARGETS_FILE and elxLIBRARY_DEPENDS_FILE. Guarding the following
-# include statements allow users to include this file directly for backwards 
-# compatibility.
+# ELASTIX_CONFIG_TARGETS_FILE.
 if( Elastix_FOUND )
```
Comment (N-Dekker) [inline UseElastix.cmake.in]: @kaspermarstal Should the _if_ be removed, now that I removed the entire _else_ clause?

File: ElastixConfig.cmake.in [hunk 2]
```diff
@@ -24,4 +24,3 @@ set( ELASTIX_HELP_DIR @ELASTIX_HELP_DIR@ )
 
 # Maintain backwards compatibility by also exporting old-style target information
 set( ELASTIX_ALL_COMPONENT_LIBS @AllComponentLibs@ )
```
Comment (N-Dekker) [inline ElastixConfig.cmake.in]: @kaspermarstal Should this line with ELASTIX_ALL_COMPONENT_LIBS also be removed? But if so, what about @AllComponentLibs@?

File: UseElastix.cmake.in (hunk 1 above)
Comment (kaspermarstal) [reply UseElastix.cmake.in]: Yes!

File: ElastixConfig.cmake.in (hunk 2 above)
Comment (kaspermarstal) [reply ElastixConfig.cmake.in]: Everything allcomponentlibs-related should be removed :)

Comment (N-Dekker) [reply ElastixConfig.cmake.in]: Also this one? https://github.com/SuperElastix/elastix/blob/develop/CMakeLists.txt#L562

File: UseElastix.cmake.in (hunk 1 above)
Comment (N-Dekker) [reply UseElastix.cmake.in]: Marius wondered if there shouldn't be an else-clause, producing an error message when Elastix is not found. But you think that's not necessary, right?

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,17 +1,9 @@
-cmake_minimum_required( VERSION 2.8 )
+cmake_minimum_required( VERSION 3.5.1 )
```
Comment (kaspermarstal) [inline CMakeLists.txt]: Is this strictly necessary? If so, fine, if not, it might be annoying to users to work around.

**Comment by kaspermarstal** [comment]
We could rebase this branch on develop and see if it works on azure.

**Comment by N-Dekker** [comment]
The CMake warnings are fixed and the old style export of the elastix library has been dropped by other pull requests:
 
- Pull request #282 "COMP: Fix CMake 3.x warnings policy CMP0033 and CMP0007 (issue #21)"
- Pull request #263 "ENH: Upgrade cmake_minimum_required from 2.8 to 3.10.2"

It is unclear whether the SuperElastix Linux link errors (https://github.com/SuperElastix/elastix/issues/21#issuecomment-405554289) would still occur with the latest elastix library version. But we have no clue and no resources to address them any time soon. This pull request (#46) is obsolete anyway now.

**Pull request closed without merging**